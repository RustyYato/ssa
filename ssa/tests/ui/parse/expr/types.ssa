let x: (addr, u32, i32, f32, f64, i43, u52, std::option::Option[T]) = x;
