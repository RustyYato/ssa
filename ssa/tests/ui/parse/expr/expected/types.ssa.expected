[
    id: 1
    kind:
        Let:
            binding:
                id: 1
                name: "x"
            ty:
                id: 10
                kind:
                    Tuple: [
                        id: 1
                        kind:
                            Primitive: "Addr",
                        id: 2
                        kind:
                            Primitive:
                                UInt:
                                    bits: 32,
                        id: 3
                        kind:
                            Primitive:
                                SInt:
                                    bits: 32,
                        id: 4
                        kind:
                            Primitive: "Float32",
                        id: 5
                        kind:
                            Primitive: "Float64",
                        id: 6
                        kind:
                            Primitive:
                                SInt:
                                    bits: 43,
                        id: 7
                        kind:
                            Primitive:
                                UInt:
                                    bits: 52,
                        id: 9
                        kind:
                            Concrete:
                                name:
                                    segments: [
                                        id: 2
                                        name: "std",
                                        id: 3
                                        name: "option",
                                        id: 4
                                        name: "Option"
                                    ]
                                generics: [
                                    id: 8
                                    kind:
                                        Concrete:
                                            name:
                                                segments: [
                                                    id: 5
                                                    name: "T"
                                                ]
                                            generics: []
                                ]
                    ]
            value:
                id: 1
                kind:
                    Ident:
                        id: 6
                        name: "x"
]